src/modules/oh-indicators-list/network-layer/sseClient.ts

```diff
@@ -1,18 +1,41 @@
+interface IEventHandlers {
+  onMessageHandler: (event: MessageEvent) => void;
+  onOpenHandler: () => void;
+  onErrorHandler: () => void;
+}
 export interface IEventSourceInitializer {
   init: (
-    eventName: string,
-    eventHandlers: {
-      onMessageHandler: (event: MessageEvent) => void;
-      onOpenHandler: (eventName: string) => void;
-      onErrorHandler: (e: any) => void;
-    },
+    url: string,
+    eventHandlers: IEventHandlers,
   ) => void;
 }
 
 export interface IEventSourceFinalizer {
-  destroy: (eventName: string) => void;
+  destroy: () => void;
 }
 
-export default class {
-    // TODO implementation
-}
\ No newline at end of file
+export const parseStreamingResponse = (
+  incomingMessage: MessageEvent,
+): { name: string; value: string } => {
+  const data = JSON.parse(incomingMessage.data);
+  const [, , name] = data.topic.split('/');
+  const { value } = JSON.parse(data.payload);
+  return { name, value };
+};
+
+export default class SseClient {
+  source: EventSource | undefined;
+  init(url: string, eventHandlers: IEventHandlers) {
+    this.source = new EventSource(url);
+    this.source.onmessage = event => eventHandlers.onMessageHandler(event);
+    this.source.onopen = () => {
+      eventHandlers.onOpenHandler();
+      this.source?.close()
```

Comment:
Does it close the connection immediately after opening?

Reply:
nope, set it in wrong place

Reply:
will move it to onerror